src/Agent.php

```diff
@@ -102,4 +104,17 @@ public function getRequestStartTime(): float
     {
         return $this->request_start_time;
     }
+
+    public function send(): bool
+    {
+        $sent = parent::send();
+
+        $this->collectors->each(function (EventDataCollector $collector) {
+            $collector->reset();
+        });
+
+        $this->putEvent(new Metadata([], $this->getConfig(), $this->agentMetadata()));
```

Comment:
You are adding this `Metadata` event after previous events have been sent to make it available for the next "connection" right?

Reply:
Yes. That reproduces the state of the Collector as it would immediately after the Agent is constructed. It's not what I consider an elegant solution, but it is completely within the context of this package and should work with the old philkra agent, as well as the 7.4 release of the nipwaayoni agent. I don't think I'll fix this in 7.5, but I intend to redo the event creation/collection in the next major release. I added a comment to explain why this code exists.

Reply:
@dstepe keep in mind that this fix is for the previous agent, and `agentMetadata` is not available :)